Multisig/UI/Transaction/InitiateTransaction/ReviewSafeTransactionViewController/ReviewSafeTransactionViewController.swift

```diff
@@ -149,19 +149,20 @@ class ReviewSafeTransactionViewController: UIViewController {
                     self.safeTxGas = UInt256String(estimatedSafeTxGas)
                 }
 
-                if self.shouldLoadTransactionPreview {
+                if self.shouldLoadTransactionPreview, let transaction = self.createTransaction() {
+                    self.transactionPreview = nil
+
                     self.currentDataTask = App.shared.clientGatewayService.asyncPreviewTransaction(
-                        transaction: tx,
+                        transaction: transaction,
                         sender: AddressString(self.safe.addressValue),
                         chainId: self.safe.chain!.id!
                     ) { result in
-                        switch result {
-                        case .success(let response):
-                            self.transactionPreview = response
-                            self.onSuccess()
-                        case .failure(let error):
-                            DispatchQueue.main.async { [weak self] in
-                                guard let `self` = self else { return }
+                        DispatchQueue.main.async {
```

Comment:
just for info, does this cased you any noticeable issue so far ?

Reply:
no noticeable issue, but I noticed this is called on background thread, but the onSuccess is calling UI-related code